java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -598,8 +552,81 @@ JsonSchemaFilter getRootDefinitions() {
         return rootDefinitions;
     }
 
-    private Optional<Map.Entry<Endpoint, Pattern>> getEndpoint(String httpMethod, URL url) {
-        String relativeUrl = URLUtils.relativeURL(url);
+
+    private String getTargetHostPath() {
+        if (targetHostPath == null) {
+            synchronized ($writeLock) {
+                if (targetHostPath == null) {
```

Comment:
I think the issue is that you checked that it's `null` on L557, but then thread blocks at 558 until it can acquire the write lock; once you have the write lock, must check again in case whoever had the lock wrote the value.

Thread-safety needed here, bc if we move to GCP cloud functions v2, the functions can handle multiple concurrent requests.